actors/account/src/lib.rs

```diff
@@ -34,25 +34,23 @@ pub enum Method {
 pub struct Actor;
 impl Actor {
     /// Constructor for Account actor
-    pub fn constructor<BS, RT>(rt: &mut RT, address: Address) -> Result<(), ActorError>
+    pub fn constructor<BS, RT>(rt: &mut RT, address: Address) -> Result<(), Abort>
```

Comment:
Given that `Abort` can't happen, why not just return `()`?

Reply:
Because I wanted to potential for actually returning an error in debug scenarios. My idea was that we could have sth like a `debug` feature, where `Abort` would collect data and actually be a returned value.

Reply:
One alternative design I had was 

```rust
fn foo() -> Abort<T> {}

type Abort<T> = Result<T, AbortError>;
```

to make it even more visual that this is a special kind of exit.

Reply:
> Because I wanted to potential for actually returning an error in debug scenarios. My idea was that we could have sth like a debug feature, where Abort would collect data and actually be a returned value.

Ok, that's reasonable. Although, IMO, we get _better_ information if we abort immediately (and ask wasmtime for a backtrace). See https://docs.wasmtime.dev/api/wasmtime/struct.Config.html#method.wasm_backtrace_details.

Reply:
There's some scope for divergent behaviour between the two cases here. I guess that this Abort is intended to _always_ be immediately propagated with `?`.

Up at this high level of the actor entry points, I would probably lean towards Steb's initial take that actually aborting is better. We would want a way to shim that call for testing, though, wrapping out the underlying static call so we can use a mock/fake VM instead.